docs/api.rst

```diff
@@ -6,7 +6,48 @@
 
 
 LIBBPF API
-==================
+==========
+
+Error Handling
+--------------
+
+When libbpf is used in "libbpf 1.0 mode", API functions can return errors in one of two ways.
+
+You can set "libbpf 1.0" mode with the following line:
+
+.. code-block::
+
+    libbpf_set_strict_mode(LIBBPF_STRICT_DIRECT_ERRS | LIBBPF_STRICT_CLEAN_PTRS);
+
+If the function returns an error code directly, it uses 0 to indicate success
+and a negative error code to indicate what caused the error. In this case the
+error code should be checked directly from the return, you do not need to check
+errno. 
+
+For example:
+
+.. code-block::
+
+    err = some_libbpf_api_with_error_return(...);
+    if (err < 0) {
+        /* Handle error accordingly */
+    }
+
+If the function returns a pointer, it will return NULL to indicate there was 
+an error. In this case errno should be checked for the error code.
+
+For example:
+
+.. code-block::
+
+    if (err < 0) {
```

Comment:
I think you mixed up examples. For pointer returning APIs you don't have `err` anymore and can only use errno. While the example above could show that err and errno could be used almost interchangeably (except errno and err sign differs and errno can be overriden by any glibc or libbpf API function)

Reply:
Ah my mistake, fixed that. 

In the case of non-pointer API functions we could document that errno and the returned code could be used interchangeably. I don't think we should though, better to keep it simple. wdyt?